### PR TITLE
added Role entity, Dao and Service layer for it and Configure authent…

### PR DESCRIPTION
**Pull request by KostiantynVorobiov**
…ication by roles

### CODE REVIEW

File: src/main/java/com/cinema/model/Role.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Enumerated(EnumType.STRING)
+    private RoleName roleName;
+
+    public Role() {
+    }
+
+    public Role(RoleName valueOf) {
```
Comment (OleksiiTromsa) [inline src/main/java/com/cinema/model/Role.java]:
```suggestion

    private Role(RoleName valueOf) {
```

File: src/main/java/com/cinema/util/InjectData.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import com.cinema.service.RoleService;
+import com.cinema.service.ShoppingCartService;
+import com.cinema.service.UserService;
+import java.util.Set;
+import javax.annotation.PostConstruct;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class InjectData {
```
Comment (OleksiiTromsa) [inline src/main/java/com/cinema/util/InjectData.java]:
```suggestion
public class DataInjector {
```
Class name should be a noun

**Review by OleksiiTromsa — COMMENTED** [review]
Good job!

File: src/main/java/com/cinema/model/Role.java (hunk 1 above)
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/model/Role.java]: why variable is called valueOf?

File: src/main/java/com/cinema/security/CustomUserDetailService.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+public class CustomUserDetailService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        User user = userService.findByEmail(email).get();
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/security/CustomUserDetailService.java]: use here orElseThrow() with UsernameNotFoundException

File: src/main/java/com/cinema/util/InjectData.java (hunk 2 above)
Comment (RomanDubovskyi) [reply src/main/java/com/cinema/util/InjectData.java]: > Class name should be a noun
\+

Comment (KostiantynVorobiov) [reply src/main/java/com/cinema/util/InjectData.java]: Done

File: src/main/java/com/cinema/model/Role.java (hunk 1 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/cinema/model/Role.java]: Done

File: src/main/java/com/cinema/security/CustomUserDetailService.java (hunk 3 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/cinema/security/CustomUserDetailService.java]: Done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**